src/decide.ts

```diff
@@ -108,12 +120,21 @@ async function assessCandidateHelper(
 	if (hashesToExclude.includes(candidateMeta.infoHash)) {
 		return { decision: Decision.INFO_HASH_ALREADY_EXISTS };
 	}
-
-	if (!compareFileTrees(candidateMeta, searchee)) {
+	const { dataDirs, dataMode } = getRuntimeConfig();
+	const perfectMatch = compareFileTrees(candidateMeta, searchee);
+	if (perfectMatch) {
+		return { decision: Decision.MATCH, metafile: candidateMeta};
+	}
+	if (!dataDirs || dataDirs.length == 0) {
 		return { decision: Decision.FILE_TREE_MISMATCH };
 	}
-
-	return { decision: Decision.MATCH, metafile: candidateMeta };
+	if (!statSync(searchee.path).isDirectory() && 
+		compareFileTreesIgnoringNames(candidateMeta, searchee) &&
+		dataMode == "risky") {
+		return { decision: Decision.MATCH_EXCEPT_PARENT_DIR, metafile: candidateMeta};
```

Comment:
is it really a MATCH_EXCEPT_PARENT_DIR if every file comparison just checked for size, not name?